### PR TITLE
[Path] Feature/camotics

### PR DESCRIPTION
**Pull request by sliptonic**
Adds a new integration with Camotics for simulation.  Requires Camotics 1.2.2 to be installed separately.
Will check for Camotics and, if found, add a button to the toolbar.  Button becomes enabled if a Path job is selected.
Pressing will open a task panel and prepare the simulation.  User can drag the slider to see the simulation at any point.
User can also launch camotics in a separate window.  This will build a camotics project from the Job.  From here the user can use all camotics features.

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [497519330](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/497519330) was triggered at [a84b2e6](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/a84b2e66e3bfe4f1b95ca788a41a195fc9e6bfbe). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [503110196](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/503110196) was triggered at [6a17f8d](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/6a17f8d5789197a5934c96913b43eb5e2e07b182). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [504494908](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/504494908) was triggered at [9f1f324](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/9f1f3243e81b8571583f816d733bdd360c2c21e7). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Path/InitGui.py [hunk 1]
```diff
@@ -158,7 +159,13 @@ def Initialize(self):
 
         if PathPreferences.advancedOCLFeaturesEnabled():
             try:
-                import ocl
+                subprocess.call(["camsim", "-v"])
```
Comment (agren) [inline src/Mod/Path/InitGui.py:162]: This is checking if camsim exist but i can't see camsim beeing used elsewhere. Perhaps calling camotics --version instead?

File: src/Mod/Path/PathTests/TestPathPost.py [hunk 2]
```diff
@@ -152,3 +154,317 @@ def testSplitArcs(self):
...
+        # check that the test file is structured correctly
+        self.assertTrue(len(self.job.Tools.Group) == 2)
+        self.assertTrue(len(self.job.Fixtures) == 2)
+        self.assertTrue(len(self.job.Operations.Group) == 3)
+
+        self.job.SplitOutput = False
+        self.job.OrderOutputBy = "Operation"
+
+    def test010(self):
+        # check that function returns correct hash
```
Comment (agren) [inline src/Mod/Path/PathTests/TestPathPost.py:202]: Does this comment reflect the functionality of the test? Seems like the test validate the structure of what buildPostList returns.

File: src/Mod/Path/InitGui.py (hunk 1 above)
Comment (sliptonic) [reply src/Mod/Path/InitGui.py:162]: Great point.  Yes, it's from an earlier version where I was using camsim.  I'll change it.

File: src/Mod/Path/PathTests/TestPathPost.py (hunk 2 above)
Comment (sliptonic) [reply src/Mod/Path/PathTests/TestPathPost.py:202]: Will change.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [556652237](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/556652237) was triggered at [4755328](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/47553287a738d967421a25f06c51a55e901bc3bd). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [567472752](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/567472752) was triggered at [e0e5d66](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/e0e5d66647597f6af2dc8f8c086df84e692e9ca5). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637">pipeline status</a> for feature branch [PR_6637](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6637). Pipeline [568349920](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/568349920) was triggered at [8dad6cf](https://github.com/FreeCAD/FreeCAD/pull/6637/commits/8dad6cf84d24508dd09477f0593de6be92d13221). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by bongiben** [comment]
How can we install Camotics 1.2.2 instructions or link to where to download

not on their gtihub from what I can see

**Pull request merged**